lib/prune.js

```diff
@@ -21,6 +21,8 @@ prune.completion = require('./utils/completion/installed-deep.js')
 
 function prune (args, cb) {
   var dryrun = !!npm.config.get('dry-run')
+  // prune should not touch the lock file
+  npm.config.set('save', false)
```

Comment:
I think the proper config option to use for this is `shrinkwrap`. The package-lock.json file is only not produced as a side-effect of setting `save` to `false`.